Makefile

```diff
@@ -58,19 +58,23 @@ $(ENV_DIR)/bin/python:
 $(ENV_DIR)/bin/coverage: $(ENV_DIR)/bin/python
 	$(ENV_DIR)/bin/pip install coverage
 
-.PHONY: build_common_egg
-build_common_egg: $(ENV_DIR)/bin/python setup.py libclearwaterutils.a
-	$(COMPILER_FLAGS) ${ENV_DIR}/bin/python setup.py bdist_egg -d $(EGG_DIR)
+.PHONY: build_common_wheel
+build_common_wheel: $(ENV_DIR)/bin/python setup.py libclearwaterutils.a
+	$(COMPILER_FLAGS) ${ENV_DIR}/bin/pip wheel -w ${WHEELHOUSE} -r requirements.txt .
 
-${ENV_DIR}/.eggs_installed : $(ENV_DIR)/bin/python setup.py $(shell find metaswitch -type f -not -name "*.pyc") libclearwaterutils.a
-	# Generate .egg files for python-common
-	$(COMPILER_FLAGS) ${ENV_DIR}/bin/python setup.py bdist_egg -d .eggs
+${ENV_DIR}/.wheels_installed : $(ENV_DIR)/bin/python setup.py requirements.txt $(shell find metaswitch -type f -not -name "*.pyc") libclearwaterutils.a
+	rm -rf .wheelhouse
 
-	# Download the egg files they depend upon
-	${ENV_DIR}/bin/easy_install -zmaxd .eggs/ .eggs/*.egg
+	# Generate .whl files for python-common and dependencies
+	$(COMPILER_FLAGS) ${ENV_DIR}/bin/pip wheel -w .wheelhouse -r requirements.txt .
```

Comment:
This line is almost the same as the `build_common_wheel` target. Why are they so similar but different? Shouldn't we use the wheels in the `WHEELHOUSE` for install and test?

Reply:
Ah, I see. It's not clear to me from this Makefile how we expect `build_common_wheel` to be used. I think it should be possible to commonize some code here?

Reply:
I've tweaked the above to just build itself, whereas this needs to build the wheel and be ready to install all of it's dependencies.